### PR TITLE
Adds Notification molecule

### PR DESCRIPTION
**Pull request by anselmbradford**
## Additions

- Adds Notification molecule

## Testing

- Visit `/browse-filterable/` 

## Review

- @KimberlyMunoz 
- @jimmynotjim 
- @sebworks 

## Screenshots

screen shot 2015-12-09 at 10 18 42 am

## Todos

- https://github.com/cfpb/cfgov-refresh/pull/1185 should be merged in first.

### CODE REVIEW

**Comment by sebworks** [comment]
LGTM

File: cfgov/jinja2/v1/_includes/molecules/notification.html [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{% macro render(options={}) -%}
+    {% set type  =  options.type if options.type in ['success', 'warning', 'error'] else 'success' %}
+    {% set icon_lookup  = {
+          'success' : 'approved',
+          'warning' : 'error',
+          'error'   : 'delete'
+        }
+    %}
+
+    {% if type == 'success' %}
```
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Might be able to make the lookup a tuple like ` success: ('approved', 'Success') `

File: cfgov/jinja2/v1/_includes/molecules/notification.html [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {% set notification_type  =  ('m-notification__' + type) if type %}
+    {% set notification_icon_type  =  (' m-notification_icon__' + type) if type %}
+    {% set cf_icon_style           =  (' cf-icon-' + icon_lookup[type] + '-round') if type %}
+    <div class="m-notification
+                {{ notification_type }}">
+        <span class="m-notification_icon
+                     {{ notification_icon_type }}
+                     cf-icon
+                     {{ cf_icon_style }}"></span>
+        <div class="m-notification_message">{{ message }}</div>
```
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: could you just make this {{ `options.message` or message }} ?

**Comment by ascott1** [comment]
It would be _suuuuuper cool_ if these were in Capital Framework. hinthintnudgenudge

**Comment by anselmbradford** [comment]
> It would be suuuuuper cool if these were in Capital Framework. hinthintnudgenudge

@ascott1 Sorry to lazyweb this, but if these were in Capital Framework, which CF repo should they be added to?

**Comment by jimmynotjim** [comment]
There's already a cf-notifications repo, I just never finished the initial work due to time :(

File: cfgov/jinja2/v1/_includes/molecules/notification.html [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    {% if type == 'success' %}
+        {% set message  =  options.message | default('Success.') %}
+    {% elif type == 'warning' %}
+        {% set message  =  options.message | default('Warning!') %}
+    {% elif type == 'error' %}
+        {% set message  =  options.message | default('Error!') %}
+    {% endif %}
+
+    {% set notification_type  =  ('m-notification__' + type) if type %}
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: There's a lot of extra white space around the `=` here and below that could be removed.

File: cfgov/jinja2/v1/browse-filterable/index.html [hunk 4]
```diff
@@ -15,6 +16,30 @@
...
+        {{ notification.render({
+            'type': 'warning'
+        }) }}
+
+        {{ notification.render({
+            'type': 'error'
+        }) }}
+
+        {{ notification.render({
+            'message': 'Custom text',
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/browse-filterable/index.html]: Left align values here and below

File: cfgov/jinja2/v1/browse-filterable/index.html [hunk 5]
```diff
@@ -15,6 +16,30 @@
...
+            'explanation': 'With an explanation!'
+        }) }}
+
+        {{ notification.render({
+            'type': 'warning',
+            'message': 'Custom text',
+            'explanation': "And an explanation with a <a href='#'>link</a>" | safe
```
Comment (jimmynotjim) [inline cfgov/jinja2/v1/browse-filterable/index.html]: Use single quotes for consistency

File: cfgov/unprocessed/css/molecules/notification.less [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+@m-notification-padding__px: 15px;
+
+.m-notification {
+    /* display: none; TODO: uncomment this line when prototype is merged into filters. x*/
+    position: relative;
+
+    padding: @m-notification-padding__px;
+    padding-left: 40px;
+    margin-bottom: @m-notification-padding__px + 4px;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/notification.less]: What's this value from? Was it a visual adjustment?

File: cfgov/unprocessed/css/molecules/notification.less [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            color: @gold;
+        }
+
+        &__error {
+            color: @redorange;
+        }
+    }
+
+    &_message {
+        .h4();
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/notification.less]: It's probably better to use `.heading-4()`

File: cfgov/unprocessed/css/molecules/notification.less [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        margin: 0;
+    }
+
+    &_explanation {
+        .h4();
+        display: block;
+        margin: 0;
+        margin-top: unit( (@grid_gutter-width / 2) / @base-font-size-px, em);
+        font-size: @base-font-size-px;
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/notification.less]: Convert to `font-size` to `em`s

File: cfgov/jinja2/v1/browse-filterable/index.html (hunk 5 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/browse-filterable/index.html]: This is a nested quote, I thought we used double when the quote contains a single quote? Or are we escaping the inside quotes `\'`?

Comment (jimmynotjim) [inline cfgov/jinja2/v1/browse-filterable/index.html]: I think we've used double quotes around html property values and kept single quotes around everything

File: cfgov/jinja2/v1/_includes/molecules/notification.html (hunk 3 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/6e40079dacf78a192e5d1d74371155c22fb54e9e

File: cfgov/jinja2/v1/browse-filterable/index.html (hunk 4 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/browse-filterable/index.html]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/9102c4a290dfa97405c249c2165b547e7aa1df10

File: cfgov/jinja2/v1/browse-filterable/index.html (hunk 5 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/browse-filterable/index.html]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/9102c4a290dfa97405c249c2165b547e7aa1df10

File: cfgov/unprocessed/css/molecules/notification.less (hunk 6 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/notification.less]: Hm, this actually should be there at all. A notification shouldn't be aware of its margin. Removed in https://github.com/cfpb/cfgov-refresh/commit/e1a7a023f616f09735e4038ab6a9ba8c50cfcbc6

File: cfgov/unprocessed/css/molecules/notification.less (hunk 8 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/notification.less]: Edit added in https://github.com/cfpb/cfgov-refresh/commit/3a0bec0089e84b50c83c9ac9fe0da8f46e6c85b2

Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/notification.less]: Cleaned up via feedback in chat in https://github.com/cfpb/cfgov-refresh/commit/0ac8f6ab861acc9131c593372db459cc4f636394

File: cfgov/jinja2/v1/_includes/molecules/notification.html (hunk 1 above)
Comment (jimmynotjim) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Ditto this.

File: cfgov/unprocessed/css/molecules/notification.less [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    - "Markup and styling for notifications"
+  patterns:
+    - name: Default Usage
+      markup: |
+        <div class="m-notification m-notification__error">
+          <span class="cf-notification_icon
+                       cf-notification_icon__error
+                       cf-icon
+                       cf-icon-error-round"></span>
+          <div class="m-notification_message">
```
Comment (jimmynotjim) [inline cfgov/unprocessed/css/molecules/notification.less]: This needs to be updated to match your new markup.

File: cfgov/jinja2/v1/browse-filterable/index.html [hunk 10]
```diff
@@ -19,6 +20,43 @@
 
 {% block content_main %}
     <div class="block
+                block__flush
+                block__padded-bottom">
+        {# TODO: Move notification to filter-list-control. #}
```
Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/browse-filterable/index.html]: Can we demo these in the filter now?

File: cfgov/unprocessed/css/molecules/notification.less [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+*/
+
+@m-notification-padding__px: 15px;
+
+.m-notification {
+    /* display: none; TODO: uncomment this line when prototype is merged into filters. x*/
+    position: relative;
+
+    padding: @m-notification-padding__px;
+    padding-left: 40px;
```
Comment (KimberlyMunoz) [inline cfgov/unprocessed/css/molecules/notification.less]: convert to ems?

File: cfgov/unprocessed/css/molecules/notification.less [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    &__error {
+        background: @redorange-20;
+        border-color: @input-error;
+    }
+
+    &_icon {
+        position: absolute;
+        top: @m-notification-padding__px + 2px;
```
Comment (KimberlyMunoz) [inline cfgov/unprocessed/css/molecules/notification.less]: Should these be converted to ems?

File: cfgov/jinja2/v1/_includes/molecules/notification.html (hunk 1 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Not sure I'm following, what would the full code block look like?

File: cfgov/unprocessed/css/molecules/notification.less (hunk 9 above)
Comment (anselmbradford) [inline cfgov/unprocessed/css/molecules/notification.less]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/3aba3bedb96103bc945bd8755eae7f2b70f676ad

File: cfgov/jinja2/v1/_includes/molecules/notification.html (hunk 1 above)
Comment (sebworks) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Something like this:
```
{% set lookup  = {
          'success' : ('approved', 'Success'),
          'warning' : ('error', 'Warning'),
          'error'   :  ('delete', 'Error!)'
        }
    %}

{% set icon, message = lookup[type]  %}
```

 ```<p class="h4">{{ options.message || message }}</p>```

Comment (anselmbradford) [inline cfgov/jinja2/v1/_includes/molecules/notification.html]: Fixed in https://github.com/cfpb/cfgov-refresh/commit/eed72c63841678f6b7e388e0b4c5fd7955a8dd49

**Comment by sebworks** [comment]
:+1:

File: cfgov/jinja2/v1/browse-filterable/index.html (hunk 10 above)
Comment (anselmbradford) [inline cfgov/jinja2/v1/browse-filterable/index.html]: I'm working on an integration but might make sense in another PR?

Comment (KimberlyMunoz) [inline cfgov/jinja2/v1/browse-filterable/index.html]: :ok_hand:

**Comment by KimberlyMunoz** [comment]
:+1:
